src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/BitbucketPullRequestDecorator.java

```diff
@@ -74,10 +73,10 @@
             Issue.STATUSES.stream().filter(s -> !Issue.STATUS_CLOSED.equals(s) && !Issue.STATUS_RESOLVED.equals(s))
                     .collect(Collectors.toList());
 
-    private final BitbucketClient client;
+    private ThreadLocal<BitbucketClient> client = new ThreadLocal<>();
```

Comment:
The `ThreadLocal` seems unnecessary given there's no real need to hold the client state at class level. I'll make code suggestions for each change I think needs made to allow a the client to be defined at a method scope and passed around.